examples/plotting/file/line_toggle.py

```diff
@@ -0,0 +1,40 @@
+"""
+Example demonstrating the picking of line objects. The transparency of
+the lines is toggled when clicked.
+"""
+
+import numpy as np
+from bokeh.plotting import output_file, show, figure, Figure
+from bokeh.models import TapTool, CustomJS
+
+# Normally, the use of the tap tool causes Bokehjs to "select" a line
+# when it is clicked on, which is not what we want in this example. To
+# deal with this, we modify the "selected" object to make it look as
+# if no glyph was clicked.
+
+code = """
+d0 = cb_obj.get("selected")["0d"];
+if (d0.glyph) {
+    var alpha = d0.glyph.visuals.line.alpha.value();
+    var color = d0.glyph.visuals.line.color.value();
+    console.log('Toggled line', d0.glyph.id, color);
+    d0.glyph.visuals.line.alpha.fixed_value = (alpha == 1) ? 0.5 : 1.0;
+    // Make selected/unselected lines have normal appearance
+    d0.glyph['renderer'].selection_glyph = null;
+    d0.glyph['renderer'].nonselection_glyph = null;
```

Comment:
> IMO this is a pretty ok way to prevent Bokeh's internals from taking action on the line click 

What if there are other callbacks, some that happen earlier some that happen later, that all look at this information? They will see inconsistent views. It seems incredibly fragile to me. 

In any case, there are several other various ways that you use-case could be achieved. That a particular glyph has been *selected* is information, what to do with that is *policy*. I think the best solution to this would not change the selection made by the selection tool, because that's addressing the wrong cause. In your case there are a few options:

* The selection tool is for selecting, that's why it sets a selection. Instead of trying to do an end run around it's normal and intended behavior, make a new tool that does hit-testing, but doesn't set selections. The tool itself could be custom to update visuals how you like, or more generally provide a callback that could act on the hit test information in some different way than the select tool. 

* Alternatively, make a new tool that sets *multiple* selections, based off the one hit-test result. Then the standard selection/nonselection glyph machinery might work perfectly as-is. If not, if we make BokehJS respect 'None' for selection/nonselection glyphs, then a selection callback would have the information to do whatever. 

We should definitely change BokehJS to respect `None` for selection/nonselection glyphs as I described above (in another PR) then at least the current way to doing things could be made better, since you'd only have to set those glyphs once, in python, to `None` and not have the callback needlessly set the to `null` on every mouse move. 

So, I do think the PR is good, adding `glyph` is useful. But I don't think this example demonstrates a technique we should encourage.